### PR TITLE
Replace lambda with method reference in SpringBootGenerator (#3371)

### LINKED ISSUE #3371: SpringBootGenerator: Lambda can be replaced with method reference
### Description

The class [SpringBootGenerator](https://github.com/eclipse-jkube/jkube/blob/03cfc9119caf9a6dec066f3b2254b7dd185d6933/jkube-kit/jkube-kit-spring-boot/src/main/java/org/eclipse/jkube/springboot/generator/SpringBootGenerator.java#L87) presents the issue: Lambda can be replaced with method reference.

The following line:
https://github.com/eclipse-jkube/jkube/blob/03cfc9119caf9a6dec066f3b2254b7dd185d6933/jkube-kit/jkube-kit-spring-boot/src/main/java/org/eclipse/jkube/springboot/generator/SpringBootGenerator.java#L87
should be changed to:

```java
        return nestedGenerator.getEnv(super::getEnv, prePackagePhase);
```
### :red_circle: Before you start :red_circle:

:point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down:

Make sure you read the [contributing guide](https://www.eclipse.org/jkube/contributing) first.

Pay special attention to the ECA agreement section and the requirement to sign-off your commit.

This is a [first-timers-only](https://eclipse.dev/jkube/contributing/#first-timers-only-issues) issue. These issues are especially created for contributors who are contributing to the project for the **first time**.

If you have already contributed to this project in the past, please don't ask to be assigned one of these issues and look for [good-first-issues](https://github.com/eclipse-jkube/jkube/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22good+first+issue%22) or [help-wanted](https://github.com/eclipse-jkube/jkube/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22help+wanted%22+) issues instead in order to continue you contributor journey in the project.

### PR DESCRIPTION
**Pull request by Aryant-Tripathi**
Fixes #3371
- This pull request addresses issue #3371, which suggests replacing lambda expressions with method references where applicable in the SpringBootGenerator. Method references offer more concise syntax and can improve readability.

**Changes:**

- Refactored lambda expressions that can be simplified by method references.
- Updated occurrences where method references provide cleaner and more efficient code, ensuring no changes to the functionality of the SpringBootGenerator.

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/eclipse-jkube/ci)

Started new GH workflow run for https://github.com/eclipse-jkube/jkube/pull/3443 (_2024-10-09T18:03:21Z_)

:gear: [JKube E2E Tests (11260515588)](https://github.com/eclipse-jkube/ci/actions/runs/11260515588) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#3443)](https://github.com/eclipse-jkube/ci/actions/runs/11260515588/job/31311909368)

**Comment by rohanKanojia** [comment]
Could you please check why build is failing on JDK8 ?

**Comment by Aryant-Tripathi** [comment]
Yes sure, checked with the issue and got error as:  **incompatible types: invalid method reference**

Attaching screenshot for reference: 
Screenshot 2024-10-10 at 21 27 43
-------

I am wondering If we can do something like this: 
```
    @Override
    protected Map<String, String> getEnv(boolean prePackagePhase) {
        return nestedGenerator.getEnv(this::fetchEnv, prePackagePhase);
    }
    
    protected Map<String, String> fetchEnv(boolean prePackagePhase) {
        return super.getEnv(prePackagePhase);
    }
```

**Comment by Aryant-Tripathi** [comment]
@manusa @rohanKanojia Gentle ping, Do I need to proceed with suggestion ?

File: jkube-kit/jkube-kit-spring-boot/src/main/java/org/eclipse/jkube/springboot/generator/SpringBootGenerator.java [hunk 1]
```diff
@@ -84,7 +84,7 @@ public List<ImageConfiguration> customize(List<ImageConfiguration> configs, bool
 
     @Override
     protected Map<String, String> getEnv(boolean prePackagePhase) {
-        return nestedGenerator.getEnv(ppp -> super.getEnv(ppp), prePackagePhase);
+        return nestedGenerator.getEnv(super::getEnv, prePackagePhase);
```
Comment (manusa) [inline jkube-kit/jkube-kit-spring-boot/src/main/java/org/eclipse/jkube/springboot/generator/SpringBootGenerator.java:87]: Apparently, the statement is incompatible with Java 8.
Let's just remove the warning and delay the refactoring to the future:
```suggestion
      //TODO: Java8 compatibility, remove warning. Remove annotation once baseline is set to Java11+
      //noinspection Convert2MethodRef
      return nestedGenerator.getEnv(ppp -> super.getEnv(ppp), prePackagePhase);
```

**Review by manusa — CHANGES_REQUESTED** [review]
Sorry for the late reply.
The issue definition is wrong since it's incompatible with Java8.
Let's just remove the warning.

**Comment by Aryant-Tripathi** [comment]
@manusa Updated with the suggestion :)

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**